core/src/main/scala/cats/data/Ior.scala

```diff
@@ -708,6 +708,17 @@ sealed abstract class Ior[+A, +B] extends Product with Serializable {
       (a, b) => that.fold(a2 => false, b2 => false, (a2, b2) => AA.eqv(a, a2) && BB.eqv(b, b2))
     )
 
+  final def compare[AA >: A, BB >: B](that: AA Ior BB)(implicit AA: Order[AA], BB: Order[BB]): Int =
```

Comment:
I think it might be useful to have a comment with a high-level explanation of the semantics here. I think what you have makes sense as the natural way to do this, but what exactly its doing isn't necessarily clear at a glance.

Reply:
To be honest, I copied the semantics of `===` above and didn't think too much about it! I guess it's just standard practice for a covariant datatype?

Reply:
It's mostly the between-ness of `Both` that I think isn't necessarily obvious. I could imagine both `Left` and `Right` preceding `Both`, for example.

Reply:
(Sorry, by "semantics here" I meant the details of how the comparison happens, not the `AA >: A` part.)

Reply:
Oh sorry!! Yeah I had a look in Haskell and found https://hackage.haskell.org/package/these-1.1.1.1/docs/Data-These.html. It puts `Both` after `Right` so I'm happy to change this. Although I don't know how widely used (and therefore canonical) it is. In my head, I feel like `Both` is kind-of between `Left` and `Right` so the ordering makes sense to me but I'm happy to be persuaded otherwise!

Reply:
Hmm, my vote would be to follow the Haskell instance, then, but I don't have a strong opinion.

Reply:
I think there is an argument for the `These` choice in that if you think of this as an `(Option, Option)` pair with two `None`s prohibited, both `Left` and `Right` would precede `Both`.

Reply:
The Haskell instance is derived and I think not a product of deep thought, but I also don't see a compelling reason to diverge from prior art.  I think of the type as a cleaner `Either[A, (Option[A], B)]`, which `Both` after `Right` is consistent with.

Reply:
Cool, I'll change it. Thanks for the feedback! :)